spacy_transformers/_align.pyx

```diff
@@ -0,0 +1,83 @@
+# cython: infer_types=True, boundscheck=False
+from cython.operator cimport dereference as deref
+from cython.operator cimport preincrement as preinc
+from libc.stdint cimport uint32_t, int32_t, int64_t
+from libc.stdlib cimport free
+from libcpp.unordered_set cimport unordered_set
+from libcpp.vector cimport vector
+
+import numpy
+
+
+ctypedef unordered_set[uint32_t]* unordered_set_uint32_t_ptr
+
+
+cpdef get_span2wp_from_offset_mapping(span, wp_char_offsets):
+    # create a mapping of char indices to spacy token indices
+    cdef int span_idx = span[0].idx
+    cdef int span_i = span[0].i
+    cdef int char_idx, rel_token_i
+    # size is +1 so we don't have to check whether the text has a trailing space
+    char_to_sp_token = numpy.empty((len(span.text) + 1,), dtype="int32")
+    char_to_sp_token.fill(-1)
```

Comment:
Alternativelly, use `numpy.full`. It's definitely shorter, but not sure whether it's actually faster.

Reply:
`numpy.full` says: New in version 1.20.0

Reply:
Ah, wait, I read that wrong and it was just for one kwarg. I just hadn't noticed it in our code before?